### PR TITLE
Theme Refactoring

### PR DESCRIPTION
**Pull request by hericlesme**
## Description
- Theme + Style Refactoring
- Fix TypeScript Issues

### Changes (general)
- Bumped chakra version to `1.0.0-next.6`
- Removed some typescript issues (`// @ts-ignore`) 
- Converted all interfaces in types

### Refactored Components

  - [x] - Accordion
  - [x] - Alert 
  - [x] - Avatar 
  - [x] - Badge 
  - [x] - Button 
  - [x] - Checkbox 
  - [x] - CloseButton
  - [x] - Divider 
  - [x] - Editable 
  - [x] - Form
  - [x] - Heading 
  - [x] - Icon
  - [x] - Input 
  - [x] - NumberInput 
  - [x] - PinInput 
  - [x] - Kbd
  - [x] - Link
  - [x] - Skeleton
  - [x] - Menu
  - [x] - Modal
  - [x] - Popover
  - [x] - Progress
  - [x] - Radio 
  - [x] - Select
  - [x] - Slider
  - [x] - Spinner
  - [x] - Stat
  - [x] - Switch
  - [x] - Tabs
  - [x] - Tag 
  - [x] - Textarea
  - [x] - Tooltip

## Affected Dependencies
```
"@chakra-ui/core": "1.0.0-next.6"
"@chakra-ui/pin-input": "1.0.0-next.6"
"@chakra-ui/system": "1.0.0-next.6"
"@chakra-ui/theme-tools": "1.0.0-next.6"
"@chakra-ui/theme": "1.0.0-next.6"
"@chakra-ui/icons": "1.0.0-next.6"
```
## How has this been tested?
- Manual checking

## Checklist
- [x] I have followed the [Contribution Guidelines](https://github.com/OpenMined/.github/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/OpenMined/.github/blob/master/CODE_OF_CONDUCT.md)
- [x] I have commented my code following the [OpenMined Styleguide](https://github.com/OpenMined/.github/blob/master/STYLEGUIDE.md)
- [x] I have labeled this PR with the relevant [Type labels](https://github.com/OpenMined/.github/labels?q=Type%3A)

### CODE REVIEW

File: omui/src/components/checkbox/checkbox.style.ts [hunk 1]
```diff
@@ -1,34 +1,32 @@
...
-      borderColor: mode('gray.200', 'transparent')(props),
-      bg: mode('gray.200', 'whiteAlpha.300')(props),
-      color: mode('gray.500', 'whiteAlpha.500')(props)
-    }
-  };
-}
+const { Checkbox } = theme.components;
 
-const baseStyle = (props: Props) => {
+const baseStyle: BaseStyle<typeof Checkbox.register> = (props) => {
```
Comment (cereallarceny) [inline omui/src/components/checkbox/checkbox.style.ts:6]: I think I'm actually fine to use the checkbox styles in @chakra-ui/theme here instead.

File: omui/src/components/close-button/close-button.style.ts [hunk 2]
```diff
@@ -1,47 +1,29 @@
...
-    md: {
-      width: '32px',
-      height: '32px',
-      fontSize: '12px'
-    },
-    sm: {
-      width: '24px',
-      height: '24px',
-      fontSize: '10px'
+const { CloseButton } = theme.components;
```
Comment (cereallarceny) [inline omui/src/components/close-button/close-button.style.ts:4]: I'm actually fine to use the styles in @chakra-ui/theme here instead. Just ensure that `_focus` remains canceled though.

File: omui/src/components/collapse/collapse.tsx [hunk 3]
```diff
@@ -3,6 +3,5 @@ import { Collapse as ChakraCollapse } from '@chakra-ui/core';
 export const Collapse = ChakraCollapse;
 
 Collapse.defaultProps = {
-  animateOpacity: false,
-  timeout: 1000
+  animateOpacity: false
```
Comment (cereallarceny) [inline omui/src/components/collapse/collapse.tsx:6]: Why no timeout? Was this removed?

File: omui/src/components/form-control/form-control.tsx [hunk 4]
```diff
@@ -22,14 +22,18 @@ export const FormControl = ({
...
-    children.displayName || children.type.displayName
-  );
-  const isSwitch = ['Switch'].includes(
-    // @ts-ignore
-    children.displayName || children.type.displayName
-  );
+  const isGroupedChild =
+    children &&
+    ['CheckboxGroup', 'RadioGroup'].includes(
+      // @ts-ignore
```
Comment (cereallarceny) [inline omui/src/components/form-control/form-control.tsx:28]: Would be nice to get rid of these `@ts-ignore` errors if possible

File: omui/src/components/icon/icon.stories.js [hunk 5]
```diff
@@ -2,21 +2,15 @@ import React from 'react';
...
-} from '../../helpers/get-theme';
+import { themeAllPossibleColors } from '../../helpers/get-theme';
 
 export default { title: 'Components/Icon', decorators: [withKnobs] };
 
-const { size } = IconStyle.defaultProps;
-
 export const Default = () => (
   <Icon
     as={FaBeer}
```
Comment (cereallarceny) [inline omui/src/components/icon/icon.stories.js:13]: Why no size? We should allow for this prop to be changed in the story...

File: omui/src/components/menu/menu.style.ts [hunk 6]
```diff
@@ -1,86 +1,58 @@
-import { Props, mode, ComponentTheme, copy } from '@chakra-ui/theme-tools';
-import { SystemProps } from '@chakra-ui/system';
+import theme from '@chakra-ui/theme';
+import { BaseStyle, mode } from '@chakra-ui/theme-tools';
 
-import { ButtonStyle } from '../button';
+const { Menu } = theme.components;
 
-const menuListStyle = (props: Props): SystemProps => {
+const baseStyle: BaseStyle<typeof Menu.register> = (props) => {
```
Comment (cereallarceny) [inline omui/src/components/menu/menu.style.ts:6]: I'm actually fine to use the styles in @chakra-ui/theme here instead.

File: omui/src/components/menu/menu.tsx [hunk 7]
```diff
@@ -105,6 +102,7 @@ export const Menu = ({
           <MenuItem
             {...item}
             as={Submenu}
+            // @ts-ignore
```
Comment (cereallarceny) [inline omui/src/components/menu/menu.tsx:105]: Can we get around this?

File: omui/src/components/modal/modal.style.ts [hunk 8]
```diff
@@ -1,79 +1,54 @@
...
- */
-const getSize = (value: string) => ({
-  Content: { maxWidth: value }
-});
+const { Modal } = theme.components;
 
-export interface ModalProps {
-  scrollBehavior?: 'inside' | 'outside';
-}
+const baseStyle: BaseStyle<typeof Modal.register> = (props) => {
```
Comment (cereallarceny) [inline omui/src/components/modal/modal.style.ts:6]: I'm actually fine to use the styles in @chakra-ui/theme here instead.

File: omui/src/components/popover/popover.style.ts [hunk 9]
```diff
@@ -1,8 +1,11 @@
-import { ComponentTheme, mode } from '@chakra-ui/theme-tools';
+import theme from '@chakra-ui/theme';
+import { BaseStyle, mode } from '@chakra-ui/theme-tools';
 
-const Popover: ComponentTheme = {
-  baseStyle: (props) => ({
-    Content: {
+const { Popover } = theme.components;
+
+const baseStyle: BaseStyle<typeof Popover.register> = (props) => {
```
Comment (cereallarceny) [inline omui/src/components/popover/popover.style.ts:6]: I'm actually fine to use the styles in @chakra-ui/theme here instead.

File: omui/src/components/radio/radio.style.ts [hunk 10]
```diff
@@ -1,16 +1,19 @@
...
+import { CheckboxStyle } from '../checkbox';
 
-const Radio: ComponentTheme = {
-  defaultProps: CheckboxStyle.defaultProps,
-  baseStyle: (props) => ({
-    Control: {
-      ...baseStyle(props).Control,
+const { Radio } = theme.components;
+
+const baseStyle: BaseStyle<typeof Radio.register> = (props) => {
```
Comment (cereallarceny) [inline omui/src/components/radio/radio.style.ts:8]: I'm actually fine to use the styles in @chakra-ui/theme here instead.

File: omui/src/components/switch/switch.style.ts [hunk 11]
```diff
@@ -1,80 +1,31 @@
...
+const { Switch } = theme.components;
 
-const Switch: ComponentTheme = {
-  defaultProps: {
-    size: 'md',
-    colorScheme: 'blue'
-  },
-  baseStyle: (props) => ({
-    Track: {
+const baseStyle: BaseStyle<typeof Switch.register> = (props) => {
```
Comment (cereallarceny) [inline omui/src/components/switch/switch.style.ts:7]: I'm actually fine to use the styles in @chakra-ui/theme here instead.

File: omui/src/components/tabs/tabs.style.ts [hunk 12]
```diff
@@ -1,122 +1,19 @@
-import {
-  Props,
-  mode,
-  ComponentTheme,
-  StyleObject,
-  getColor
-} from '@chakra-ui/theme-tools';
+import theme from '@chakra-ui/theme';
+import { BaseStyle, getColor, mode, Variants } from '@chakra-ui/theme-tools';
```
Comment (cereallarceny) [inline omui/src/components/tabs/tabs.style.ts:2]: I'm actually fine to use the styles in @chakra-ui/theme here instead.

File: omui/src/docs/_helpers.tsx [hunk 13]
```diff
@@ -21,6 +21,7 @@ export const ColorPalette = () => {
           </Text>
           {Object.keys(colors[color]).map((num) => {
             const colorCode = `${color}.${num}`;
+            // eslint-disable-next-line
```
Comment (cereallarceny) [inline omui/src/docs/_helpers.tsx:24]: Why?

Comment (hericlesme) [reply omui/src/docs/_helpers.tsx:24]:
```
24:41  error  React Hook "useClipboard" cannot be called inside a callback. React Hooks must be called in a React function component or a custom React Hook function  react-hooks/rules-of-hooks
```

File: omui/src/components/collapse/collapse.tsx (hunk 3 above)
Comment (hericlesme) [reply omui/src/components/collapse/collapse.tsx:6]: Yes, it was removed on chakra

File: omui/src/components/icon/icon.stories.js (hunk 5 above)
Comment (hericlesme) [reply omui/src/components/icon/icon.stories.js:13]: The `IconStyle` was removed on chakra. A new numeric input (in pixels) was added to icon stories.

**Pull request merged**